lib/resource-timing.js

```diff
@@ -0,0 +1,228 @@
+      return null;
+    }
+  };
+
+  SNPRTProto.getResourcesFromOrigin = function(origin) {
+    if(window.performance.getEntriesByType) {
+      var resources = window.performance.getEntriesByType("resource");
+      var resourcesFromOrigin = [];
+      for var resource in resources {
+        var url = new URL(resources[resource].name);
+        if (url.origin === origin) {
+          resourcesFromOrigin.push(resources[resource]);
+        }
+      }
+      return resourcesFromOrigin;
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto.isBufferFull = function() {
+    if(window.performance.getEntriesByType) {
+      if(performance.onresourcetimingbufferfull) {
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return null;
+    }
+  };
+
+  SNPRTProto._nameExtend = function(name) {
+    if(name.charAt(0) == '/') {
+      return "http://" + window.location.hostname + name;
+    } else {
+      return name;
+    }
+  };
+
+  SNPRTProto.getResource = function(name) {
+    name = this._nameExtend(name);
+
+    return window.performance.getEntriesByName(name, "resource")[0]
+  };
+
+  SNPRTProto.duration = function(name, eventA, eventB, options) {
+    name = this._nameExtend(name);
+    var resource = this.getResource(name);
+    if (resource) {
+      if (resource[eventA] === 0 || resource[eventB] === 0) {
+        return false;
+      }
+      return surfnperf._roundedDuration(resource[eventB], resource[eventA], options);
+    } else {
+      return undefined;
+    }
+  };
+
+  SNPRTProto.start = function(name, options) {
+    name = this._nameExtend(name);
+
+    var resource = this.getResource(name);
+    if(resource == undefined) {
+      return undefined;
+    }
+    var start = resource.startTime;
+
+    return surfnperf._round(start, options);
+  };
+
+  SNPRTProto.end = function(name, options) {
+    name = this._nameExtend(name);
+
+    var resource = this.getResource(name);
+    if(resource == undefined) {
+      return undefined;
+    }
+    var end = resource.responseEnd;
+
+    return surfnperf._round(end, options);
+  };
+
+  SNPRTProto.getFullRequestLoadTime = function(name, options) {
+    name = this._nameExtend(name);
+
+    return surfnperf._round(this.getResource(name).duration, options);
+
+  };
+
+  SNPRTProto.getNetworkTime = function(name, options) {
+    return this.duration(name, 'fetchStart', 'connectEnd', options);
+  };
+
+  SNPRTProto.getServerTime = function(name, options) {
+    return this.duration(name, 'requestStart', 'responseEnd', options);
+  };
+
+  SNPRTProto.getBlockingTime = function(name, options) {
+    var blockingTime = 0;
+    var res = getResource(name);
+    if (res.connectEnd && res.connectEnd === res.fetchStart) {
+      blockingTime = res.requestStart - res.connectEnd;
+    } else if (res.domainLookupStart) {
+      blockingTime = res.domainLookupStart - res.fetchStart;
+    }
+    return surfnperf._round(blockingTime, options)
+  };
+}));
```

Comment:
you have to `return new SurfNPerfRT();` to make all of these functions available to `surfnperfRT`